lib/controller_patches.rb

```diff
@@ -8,8 +8,57 @@
 Rails.configuration.to_prepare do
   # Front page needs some additional info
   GeneralController.class_eval do
+
     def frontpage
-      blog
+      blog_cache("latest_blog_posts")
     end
+
+    private
+
+    def blog_cache(cache_key, expires=4.hours)
+      # set @blog_items to an empty array and return if there's no blog feed
+      if AlaveteliConfiguration::blog_feed.empty?
+        @blog_items = []
+        return
+      end
+      # nothing to do here, call the blog code and return
+      return get_blog_content unless AlaveteliConfiguration::cache_fragments
+
+      # read in the last timestamp
+      updated = read_fragment("#{cache_key}-lastupdated")
+
+      # construct the fragment key
+      @fragment_key = "#{cache_key}-#{updated}"
+
+      # if the fragment is unreadable or has reached expiry
+      # attempt to pull in the blog feed
+      if updated.nil? || !fragment_exist?(@fragment_key) ||
+         Time.now > (Time.parse(updated) + expires)
+        # keep a note of the old key
+        old_key = @fragment_key.dup
+
+        # prepare a new timestamp and fragment key
+        updated = create_timestamp
```

Comment:
when do you think the review will be completed?
on the 23rd the new law is in force so we need to know if Chiedi will be working